### PR TITLE
add comments to uwsgi config + tweak a few settings

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

`uwsgi` configuration is hard to understand (the docs are 💩 and what is a `harakiri`?!). This PR adds comments to the config file so that it is easier to know what each setting does. Also adds links to a few reference blog articles that I found useful.

### CODE REVIEW

File: engine/uwsgi.ini [hunk 1]
```diff
@@ -1,23 +1,100 @@
...
+;
+; This option should always be set to ‘on’ unless you are using the more complex “emperor” system for multi-app
+; deployments or are debugging specific behavior for which you want uWSGI to be limited.
 master=True
+
 pidfile=/tmp/project-master.pid
 http=0.0.0.0:8080
 processes=5
-harakiri=620
-max-requests=5000
```
Comment (joeyorlando) [inline engine/uwsgi.ini:10]: bumped `max-requests` down to `1000` and modified `harakiri` to `15` (setting it to 620 = 10m20s, probably not what we intended)

File: engine/uwsgi.ini [hunk 2]
```diff
@@ -1,23 +1,100 @@
...
+; pidfiles, or admin FIFOs.
+;
+; Leaving these files around can pose a problem under some circumstances, such as if a developer runs uWSGI as their
+; own user, and takes ownership of these files. If the production user doesn’t have permission to delete those files,
+; uWSGI may fail to function properly.
 vacuum=True
+
 buffer-size=65535
 http-auto-chunked=True
 http-timeout=620
```
Comment (joeyorlando) [inline engine/uwsgi.ini:60]: we may also want to update `http-timeout` to `15`, 620 here means 10m20s.

**Comment by mderynck** [comment]
👍 On for the docs.  Is there an expected improvement or behaviour change from changing the settings or did they just look outside normal ranges?

**Comment by joeyorlando** [comment]
@mderynck
- `need-app`, `die-on-term`, and `single-interpreter` (new settings) seemed like reasonable defaults. The [Bloomberg Tech article](https://www.bloomberg.com/company/stories/configuring-uwsgi-production-deployment/) gives a good explaination behind these.
- I bumped `harakiri=620` (seconds) down to `60`, as it 10m20s seems a long time to wait before considering killing the request and recycling the process. wdyt?

**Review by mderynck — APPROVED** [review]

**Pull request merged**